packages/expo-calendar/ios/CalendarModule.swift

```diff
@@ -0,0 +1,509 @@
+        throw ReminderNotFoundException(reminderId)
+      }
+      return reminderWithId
+    }
+    let reminder = EKReminder(eventStore: eventStore)
+    reminder.calendar = eventStore.defaultCalendarForNewReminders()
+
+    if let calendarId = details.calendarId {
+      let calendar = eventStore.calendar(withIdentifier: calendarId)
+      if let calendar {
+        if calendar.allowedEntityTypes.isDisjoint(with: .reminder) {
+          throw InvalidCalendarTypeException((calendarId, "reminder"))
+        }
+      } else {
+        throw CalendarIdNotFoundException(calendarId)
+      }
+      reminder.calendar = calendar
+    }
+    return reminder
+  }
+
+  private func getCalendar(from record: CalendarRecord) throws -> EKCalendar {
+    if let id = record.id {
+      guard let calendar = eventStore.calendar(withIdentifier: id) else {
+        throw CalendarIdNotFoundException(id)
+      }
+
+      if calendar.isImmutable == true {
+        throw CalendarNotSavedException(record.title)
+      }
+      return calendar
+    }
+    let calendar: EKCalendar
+    if record.entityType == .event {
+      calendar = .init(for: .event, eventStore: eventStore)
+    } else if record.entityType == .reminder {
+      calendar = .init(for: .reminder, eventStore: eventStore)
+    } else {
+      throw EntityNotSupportedException(record.entityType?.rawValue)
+    }
+
+    if let sourceId = record.sourceId {
+      calendar.source = eventStore.source(withIdentifier: sourceId)
+    } else {
+      calendar.source = record.entityType == .event ?
+      eventStore.defaultCalendarForNewEvents?.source :
+      eventStore.defaultCalendarForNewReminders()?.source
+    }
+
+    return calendar
+  }
+
+  private func getCalendar(from event: Event) throws -> EKEvent {
+    if let id = event.id {
+      guard let event = getEvent(with: id, startDate: parse(date: event.instanceStartDate)) else {
+        throw EventNotFoundException(id)
+      }
+      return event
+    }
+    guard let calendarId = event.calendarId else {
+      throw CalendarIdRequiredException()
+    }
+    guard let calendar = eventStore.calendar(withIdentifier: calendarId) else {
+      throw CalendarIdNotFoundException(calendarId)
+    }
+
+    if calendar.allowedEntityTypes.isDisjoint(with: [.event]) {
+      throw InvalidCalendarTypeException((calendarId, "event"))
+    }
+
+    let calendarEvent = EKEvent(eventStore: eventStore)
+    calendarEvent.calendar = calendar
+
+    return calendarEvent
+  }
+
+  private func getEvent(with id: String, startDate: Date?) -> EKEvent? {
+    guard let firstEvent = eventStore.calendarItem(withIdentifier: id) as? EKEvent else {
+      return nil
+    }
+
+    guard let startDate else {
+      return firstEvent
+    }
+
+    guard let firstEventStart = firstEvent.startDate, firstEventStart.compare(startDate) == .orderedSame else {
+      return firstEvent
+    }
+
+    let endDate = startDate.addingTimeInterval(2_592_000)
```

Comment:
Question: Why not take instanceEndDate from the event? Does that exist?

Reply:
We don't accept an option for an end date and the `endDate` on the event objecct can be nil so I think this is to avoid that